examples/tv-casting-app/android/App/app/src/main/jni/cpp/core/MatterCastingPlayer-JNI.cpp

```diff
@@ -44,50 +46,61 @@ namespace core {
+ jobject jIdentificationDeclarationOptions)
 {
     chip::DeviceLayer::StackLock lock;
-    ChipLogProgress(AppServer, "MatterCastingPlayer-JNI::verifyOrEstablishConnection() called with a timeout of: %ld seconds",
-                    static_cast<long>(commissioningWindowTimeoutSec));
+    ChipLogProgress(AppServer, "MatterCastingPlayer-JNI::verifyOrEstablishConnection() called with a timeout of: %d seconds",
+                    static_cast<int>(commissioningWindowTimeoutSec));
 
     CastingPlayer * castingPlayer = support::convertCastingPlayerFromJavaToCpp(thiz);
     VerifyOrReturnValue(castingPlayer != nullptr, support::convertMatterErrorFromCppToJava(CHIP_ERROR_INVALID_ARGUMENT));
 
-    matter::casting::core::IdentificationDeclarationOptions idOptions;
-
-    // TODO: In the following PRs. Replace EndpointFilter Java class with IdentificationDeclarationOptions Java class.
-    matter::casting::core::EndpointFilter desiredEndpointFilter;
-    if (desiredEndpointFilterJavaObject != nullptr)
+    // Find the ConnectionCallbacks class, get the field IDs of the connection callbacks and extract the callback objects.
+    jclass connectionCallbacksClass = env->GetObjectClass(jconnectionCallbacks);
+    VerifyOrReturnValue(
+        connectionCallbacksClass != nullptr, support::convertMatterErrorFromCppToJava(CHIP_ERROR_INVALID_ARGUMENT),
+        ChipLogError(AppServer, "MatterCastingPlayer-JNI::verifyOrEstablishConnection() connectionCallbacksClass == nullptr "));
+
+    jfieldID successCallbackFieldID =
+        env->GetFieldID(connectionCallbacksClass, "onSuccess", "Lcom/matter/casting/support/MatterCallback;");
+    jfieldID failureCallbackFieldID =
+        env->GetFieldID(connectionCallbacksClass, "onFailure", "Lcom/matter/casting/support/MatterCallback;");
+    jfieldID commissionerDeclarationCallbackFieldID =
+        env->GetFieldID(connectionCallbacksClass, "onCommissionerDeclaration", "Lcom/matter/casting/support/MatterCallback;");
+
+    jobject jSuccessCallback                 = env->GetObjectField(jconnectionCallbacks, successCallbackFieldID);
+    jobject jFailureCallback                 = env->GetObjectField(jconnectionCallbacks, failureCallbackFieldID);
+    jobject jCommissionerDeclarationCallback = env->GetObjectField(jconnectionCallbacks, commissionerDeclarationCallbackFieldID);
+
+    VerifyOrReturnValue(
+        jSuccessCallback != nullptr, support::convertMatterErrorFromCppToJava(CHIP_ERROR_INVALID_ARGUMENT),
+        ChipLogError(AppServer,
+                     "MatterCastingPlayer-JNI::verifyOrEstablishConnection() jSuccessCallback == nullptr but is mandatory "));
+    VerifyOrReturnValue(
+        jSuccessCallback != nullptr, support::convertMatterErrorFromCppToJava(CHIP_ERROR_INVALID_ARGUMENT),
+        ChipLogError(AppServer,
+                     "MatterCastingPlayer-JNI::verifyOrEstablishConnection() jSuccessCallback == nullptr but is mandatory "));
```

Comment:
Duplicated lines. Did you mean to test for the other callbacks here?

Reply:
Right, the second line is supposed to check for jFailureCallback